plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -235,4 +256,69 @@ public void testContainsSpecialCharacters()
+            throws Exception
+    {
+        try (DisposableSubContext organization = openLdapServer.createOrganization();
+                DisposableSubContext ignored = openLdapServer.createUser(organization, "alice", "alice-pass")) {
+            LdapConfig ldapConfig = new LdapConfig()
+                    .setLdapUrl(proxyLdapUrl)
+                    .setLdapConnectTimeout(new Duration(1, SECONDS))
+                    .setUserBindSearchPatterns("uid=${USER}," + organization.getDistinguishedName());
+
+            LdapAuthenticator ldapAuthenticator = new LdapAuthenticator(new JdkLdapAuthenticatorClient(ldapConfig), ldapConfig);
+            assertThatThrownBy(() -> ldapAuthenticator.createAuthenticatedPrincipal("alice", "alice-pass"))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessageMatching(".*Authentication error.*");
+
+            LdapConfig withIncreasedTimeout = ldapConfig.setLdapConnectTimeout(new Duration(30, SECONDS));
+            assertEquals(
+                    new LdapAuthenticator(new JdkLdapAuthenticatorClient(withIncreasedTimeout), withIncreasedTimeout)
+                            .createAuthenticatedPrincipal("alice", "alice-pass"),
+                    new BasicPrincipal("alice"));
+
+            LdapConfig withNoTimeoutConfigured = ldapConfig.setLdapConnectTimeout(null);
+            assertEquals(
+                    new LdapAuthenticator(new JdkLdapAuthenticatorClient(withNoTimeoutConfigured), withNoTimeoutConfigured)
+                            .createAuthenticatedPrincipal("alice", "alice-pass"),
+                    new BasicPrincipal("alice"));
+        }
+    }
+
+    @Test
+    public void testReadTimeout()
+            throws Exception
+    {
+        try (DisposableSubContext organization = openLdapServer.createOrganization();
+                DisposableSubContext group = openLdapServer.createGroup(organization);
+                DisposableSubContext alice = openLdapServer.createUser(organization, "alice", "alice-pass")) {
+            openLdapServer.addUserToGroup(alice, group);
+
+            LdapConfig ldapConfig = new LdapConfig()
+                    .setLdapUrl(proxyLdapUrl)
+                    .setLdapReadTimeout(new Duration(1, SECONDS))
+                    .setUserBindSearchPatterns("uid=${USER}," + organization.getDistinguishedName())
+                    .setUserBaseDistinguishedName(organization.getDistinguishedName())
+                    .setGroupAuthorizationSearchPattern(format("(&(objectClass=groupOfNames)(cn=group_*)(member=uid=${USER},%s))", organization.getDistinguishedName()));
+
+            LdapAuthenticator ldapAuthenticator = new LdapAuthenticator(new JdkLdapAuthenticatorClient(ldapConfig), ldapConfig);
+            assertThatThrownBy(() -> ldapAuthenticator.createAuthenticatedPrincipal("alice", "alice-pass"))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessageMatching(".*Authentication error.*");
```

Comment:
There is no difference as in tests above. As I understood you before authentication error is raised on connection timeout. Here I was expecting some timeout exception.

Reply:
But LDAPClient doesn't throw much of a granular exception, so I guess it would throw something similar to `Authentication error`